File: src/attribute-info.ts
export default class AttributeInfo{
    name: string
    position: number
    type: AttributeType
    buffer: WebGLBuffer
    mapper: (webGlContext: WebGLRenderingContext, position: number) => void
    constructor(
        name: string,
        position: number,
        type: AttributeType,
        buffer: WebGLBuffer,
        mapper: (webGlContext: WebGLRenderingContext, position: number) => void
    ){
        this.name = name
        this.position = position
        this.type = type
        this.buffer = buffer
        this.mapper = mapper
    }
}
export enum AttributeType{
    ARRAY_BUFFER = 34962
}

File: src/sgl/AttributeInfo.kt
package sgl
import org.khronos.webgl.WebGLBuffer
import org.khronos.webgl.WebGLRenderingContext

typealias AttributeMapper = (renderingContext: WebGLRenderingContext, position: Int) -> Unit

enum class AttributeType(val value: Int){ ARRAY_BUFFER(34962)}

data class Attribute(val name: String, val type: AttributeType, val mapper: AttributeMapper)

data class AttributeInfo(
        val name: String,
        val position: Int,
        val type: AttributeType,
        val buffer: WebGLBuffer,
        val mapper: AttributeMapper
)